src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -7,6 +7,7 @@
 import com.atlassian.bitbucket.jenkins.internal.model.BitbucketProject;
 import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
 import com.atlassian.bitbucket.jenkins.internal.model.RepositoryState;
+import com.google.common.annotations.VisibleForTesting;
```

Comment:
I know I'm late, but uurgh. 
This adds a dependency on guava, which we do not want. Further to that, it is a transitive dependency as you did not add this dependency to the pom.xml. 
I also really do not like the concept of `VisibleForTesting` it is a smell and having looked at the code I see why you did it this way, but I wonder if we can do it better?
How about we actually add a public method `List<RemoteConfig> getRemoteConfigs()`? we should add a  `@CheckForNull` on it as the remote config is `transient` which means it will not be there on remote slaves, nor on config read from disk.

Reply:
I have raised - https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/83 to fix this. We should not use guava for `visible for testing`.